tests/README.md

```diff
@@ -0,0 +1,122 @@
+
+#### text-4
+
+[run-1](./20_4_2c1.txt). No need to repeat, as same result found. It fails ❌ as it falls into a different bucket than bucket with key: `1110`. Given its 4 bits, the hashes are different by 1 bit.
+
+```txt
+original text: 1110
+modified text: 1100
+```
+
+### Type-3: AI-generated text
+
+Parse an AI-generated version of a query text taken out of the text list & check if the text falls into the pre-categorized bucket.
+
+> Used the following prompt in GPT-4 for this.
+
+```txt
+Create a text with same semantic meaning as ".....".
+```
+
+Below are the results with openai small/large embedding models.
+
+#### text-0
+
+- <u>`{bucketing: small, query: small}`</u>: [run-1](./40_4_3a.txt). It fails ❌ as it falls into a different bucket than a bucket containing 'text-0'. Also, the hash is slightly different from the original text.
+
+  ```txt
+  original text hash: 1001
+  generated text hash: 1000
+  ```
+
+  It seems like we need to switch to a better embedding model with more embedding values like `text-embedding-3-large` from available [OpenAI embedding models](https://platform.openai.com/docs/guides/embeddings/embedding-models).
+
+- <u>`{bucketing: large, query: large}`</u>: When large model used for both, the hashes were different. [run-3](./40_4_3a3.txt) ✅
+
+- <u>`{bucketing: large, query: small}`</u> [run-2](./40_4_3a2.txt) ❌
+
+#### text-4
+
+- <u>`{bucketing: small, query: small}`</u>: [run-1](./40_4_3b.txt). It fails ❌ as it falls into a different bucket than a bucket containing 'text-4'. Also, the hash is very different from the original text.
+
+```txt
+original text hash: 1110
+generated text hash: 0010
+```
+
+- <u>`{bucketing: large, query: large}`</u>: When large model used for both, the hashes were different. [run-2](./40_4_3b2.txt) ❌
+
+```txt
+original text hash: 0010
+generated text hash: 1010
+```
+
+- <u>`{bucketing: large, query: small}`</u> [run-2](./40_4_3b3.txt) ✅
+
+#### text-9
+
+- <u>`{bucketing: small, query: small}`</u>: [run-1](./40_4_3c.txt). It fails ❌ as it falls into a different bucket than a bucket containing 'text-9'. Also, the hash is very different from the original text.
+
+```txt
+original text hash: 1000
+generated text hash: 1100
+```
+
+- <u>`{bucketing: large, query: large}`</u>: [run-2](./40_4_3c2.txt) ❌. The hashes differs by 1 bit.
+
+```txt
+original text hash: 0011
+generated text hash: 0001
+```
+
+- <u>`{bucketing: large, query: small}`</u>: [run-3](./40_4_3c3.txt) ❌. The hashes are completely different.
+
+```txt
+original text hash: 0011
+generated text hash: 1100
+```
+
+## Conclusion
+
+Had to also consider openai large embedding model as well. In some cases, tests passed as in the query text fell into the expected bucket.
+At this point it’s a bit ambiguous to jump to a conclusion. There are 2 main observations as of now:
+
+1. In cases where the semantic hash is different for the original & query texts, embedding model needs to be replaced.
```

Comment:
I don't think this is true, we lose a lot of information from the original vector embedding to the hash. I don't think it will make a huge difference if you use an improved embedding model.

Reply:
Commented this thought as of now.